dask/dataframe/core.py

```diff
@@ -1178,6 +1178,64 @@ def repartition(
         elif freq is not None:
             return repartition_freq(self, freq=freq)
 
+    def shuffle(
+        self,
+        index,
```

Comment:
```suggestion
        on,
```

What do you think about this name? `index` is a bit overloaded.